### PR TITLE
Fixes for Trainer the Trainer workshop surveys

### PR DESCRIPTION
**Pull request by dmcavoy**
When investigating how best to support Train the Trainer workshops this year I noticed that the surveys for the Subject 'District' under 'CS Fundamentals' Course in the Workshop Dashboard were not working. This is the result of a bunch of special casing that was put in to determine if a survey should use Jotform or Foorm. This special casing could be improved to make the default for any new subject added be that it uses Foorm but I am not tackling that as part of this work (tracked in https://codedotorg.atlassian.net/browse/PLAT-1911). 

### Teachers Can Take Survey

https://user-images.githubusercontent.com/208083/177620914-645650a3-02f6-406c-aba6-077096ebac50.mov

### Facilitators Can View Survey

https://user-images.githubusercontent.com/208083/177605028-ccbce369-875d-415e-a251-11cbb55c9c4f.mov

### Don't show funded input for district workshops

Also during the review of the plan for train the trainer workshops I noticed that it was asking if the workshop was Code.org funded. Talking to Danielle they do not need that field. We have now removed that field.

https://user-images.githubusercontent.com/208083/177609256-d294f183-fe37-402e-9e4d-800ba9be2b6d.mov

## Links

https://codedotorg.atlassian.net/browse/PLAT-1896
https://codedotorg.atlassian.net/browse/PLAT-1904
[Plan for Train the Trainer Pilot Workshops](https://docs.google.com/document/d/1ofEeJAbIGmkNYd2k1aZxVBEMDrW-XRedl4PeMEcBNaI/edit#heading=h.se19j09d98yj)

## Testing story

- Manual testing
- Added some unit tests but did not add tests for workshopForm since that would have taken a lot more time and this change is needed quickly

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx [hunk 1]
```diff
@@ -371,7 +372,12 @@ export class WorkshopForm extends React.Component {
       isCsf &&
       this.state.subject &&
       !HideOnWorkshopMapSubjects.includes(this.state.subject);
-    const showFundedInput = !isAdminCounselor;
+    const showFundedInput = !(
+      isAdminCounselor ||
+      (isCsf &&
+        this.state.subject &&
+        HideFundedSubjects.includes(this.state.subject))
```
Comment (megcrenshaw) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx]: I find this hard to read with the not being outside the || and the && ...

Any chance you could either add a comment, or break up the conditional so it's easier to read?

File: apps/src/code-studio/pd/workshop_dashboard/workshop_summary_utils.js [hunk 2]
```diff
@@ -7,15 +7,16 @@ import {CSF, CSD, CSP, CSA} from '../application/ApplicationConstants';
...
   return (
     (lastSessionDate >= new Date('2020-05-08') &&
       (subject === SubjectNames.SUBJECT_SUMMER_WORKSHOP ||
         subject === SubjectNames.SUBJECT_CSF_101 ||
         subject === SubjectNames.SUBJECT_CSP_FOR_RETURNING_TEACHERS ||
         AcademicYearWorkshopSubjects.includes(subject))) ||
     (lastSessionDate >= new Date('2020-09-01') &&
-      subject === SubjectNames.SUBJECT_CSF_201)
+      subject === SubjectNames.SUBJECT_CSF_201) ||
+    subject === SubjectNames.SUBJECT_CSF_DISTRICT
```
Comment (megcrenshaw) [inline apps/src/code-studio/pd/workshop_dashboard/workshop_summary_utils.js:19]: Good catch!

File: apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx [hunk 3]
```diff
@@ -756,7 +760,10 @@ export default class EnrollForm extends React.Component {
 
     if (this.props.workshop_course === CSF) {
       requiredFields.push('role', 'grades_teaching');
-      if (this.props.workshop_subject === INTRO) {
+      if (
+        this.props.workshop_subject === INTRO ||
+        this.props.workshop_subject === SubjectNames.SUBJECT_CSF_DISTRICT
```
Comment (megcrenshaw) [inline apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx]: Nit: could make this a constant at the top to be consistent with the `INTRO` and `CSF` (although I realize there are other `SubjectNames.` that haven't been extracted)

File: dashboard/app/models/pd/enrollment.rb [hunk 4]
```diff
@@ -172,7 +172,7 @@ def self.filter_for_survey_completion(enrollments, select_completed = true)
     foorm_enrollments, other_enrollments = enrollments.partition do |enrollment|
       (enrollment.workshop.workshop_ending_date >= Date.new(2020, 5, 8) &&
         (enrollment.workshop.csf_intro? || enrollment.workshop.local_summer? || enrollment.workshop.csp_wfrt?)) ||
-        (enrollment.workshop.workshop_ending_date >= Date.new(2020, 9, 1) && enrollment.workshop.csf_201?)
+        (enrollment.workshop.workshop_ending_date >= Date.new(2020, 9, 1) && enrollment.workshop.csf_201?) || enrollment.workshop.csf_district?
```
Comment (megcrenshaw) [inline dashboard/app/models/pd/enrollment.rb:175]: No need for a workshop_ending_date here?  When was it switched to `foorm`? Also, could you update the comment above this line?

File: dashboard/test/models/pd/enrollment_test.rb [hunk 5]
```diff
@@ -134,6 +137,7 @@ class Pd::EnrollmentTest < ActiveSupport::TestCase
 
     studio_url = ->(path) {CDO.studio_url(path, CDO.default_scheme)}
     assert_equal studio_url["/pd/workshop_survey/csf/post101/#{csf_enrollment.code}"], csf_enrollment.exit_survey_url
+    assert_equal studio_url["/pd/workshop_survey/csf/post101/#{csf_enrollment.code}"], csf_district_enrollment.exit_survey_url
```
Comment (megcrenshaw) [inline dashboard/test/models/pd/enrollment_test.rb]: 🥳

**Review by megcrenshaw — APPROVED** [review]
Thanks for figuring this out! Sadly a lot of work needed here ... thanks for creating the follow-up too. Made some small nits, but looks good to me overall!

**Comment by dmcavoy** [comment]
Code Review V2 UI Test continues to be flaky. This was reported by other developers and I'm going to merge this to unblock these workshops.

**Pull request merged**